src/main/java/hobbylist/model/date/Date.java

```diff
@@ -49,7 +49,13 @@ public String getOrginString() {
      * Returns true if a given string is a valid tag name.
      */
     public static boolean isValidDateString(String test) {
-        return test.matches(VALIDATION_REGEX);
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
+        }
+        if (Integer.valueOf(test.split("-")[0]) < 1000) {
```

Comment:
This might work as well.
```suggestion
        if (Integer.valueOf(test.split("-")[0]) != 0) {
```